enhancements/cert-manager/istio-csr-controller.md

```diff
@@ -0,0 +1,879 @@
+	// +kubebuilder:default:="9402"
+	// +optional
+	Port string `json:"port,omitempty"`
+
+	// MetricServiceConfig is for confuring the metrics server specifics.
+	// +optional
+	MetricServiceConfig `json:"metricServiceConfig,omitempty"`
+}
+
+// MetricServiceConfig is for confuring the metrics server specifics.
+type MetricServiceConfig struct {
+	// Enabled when set will create a Service resource to expose metrics endpoint.
+	// +kubebuilder:default:=false
+	// +optional
+	Enabled bool `json:"enabled,omitempty"`
+
+	// ServiceType of the service resource created metrics.
+	// +kubebuilder:default:="ClusterIP"
+	// +optional
+	ServiceType corev1.ServiceType `json:"serviceType,omitempty"`
+
+	// MetricsServiceMonitorConfig is for configuring the service monitor specifics
+	// for the metrics service.
+	// +optional
+	MetricsServiceMonitorConfig `json:"metricsServiceMonitorConfig,omitempty"`
+}
+
+// MetricsServiceMonitorConfig is for configuring the service monitor specifics
+// for the metrics service.
+type MetricsServiceMonitorConfig struct {
+	// Enabled when set creates Prometheus ServiceMonitor resource for approver-policy.
+	// +kubebuilder:default:=false
+	// +optional
+	Enabled bool `json:"enabled,omitempty"`
+
+	// PrometheusInstance name to set the value for the "prometheus" label on the ServiceMonitor.
+	// +kubebuilder:default:="default"
+	// +optional
+	PrometheusInstance string `json:"prometheusInstance,omitempty"`
+
+	// Interval defines periodicity the Prometheus will scrape for metrics.
+	// +kubebuilder:default:="10s"
+	// +optional
+	Interval time.Duration `json:"interval,omitempty"`
+
+	// ScrapeTimeout defines timeout on each metric probe request.
+	// +kubebuilder:default:="5s"
+	// +optional
+	ScrapeTimeout time.Duration `json:"scrapeTimeout,omitempty"`
+
+	// Labels to add to the ServiceMonitor resource.
+	// +optional
+	Labels map[string]string `json:"labels,omitempty"`
+}
+
+// CertManagerConfig is for configuring cert-manager specifics.
+type CertManagerConfig struct {
+	// IssuerRef contains details to the referenced object used for
+	// obtaining the certificates.
+	// When unset operator will create Issuer in the configured istio's
+	// namespace.
+	// +kubebuilder:validation:XValidation:rule="oldSelf == '' || self == oldSelf",message="issuerRef is immutable once set"
+	// +optional
+	IssuerRef certmanagerv1.ObjectReference `json:"issuerRef,omitempty"`
+}
+
+// TLSConfig is for configuring certificate specifics.
+type TLSConfig struct {
+	// TrustDomain is the cluster's trust domain.
+	// +kubebuilder:default:="cluster.local"
+	// +optional
+	TrustDomain string `json:"trustDomain,omitempty"`
+
+	// RootCAFile is for setting the file location containing the root CA which is
+	// present in the configured IssuerRef. File should be made available using the
+	// Volume and VolumeMount options.
+	// +kubebuilder:default:="/var/run/secrets/istio-csr/ca.crt"
```

Comment:
We shouldn't set the default value for this field if we do not require mounting static root CA.
```suggestion
```

Reply:
Will remove the default.

Reply:
suggestion restored.